### DISCUSSION TITLE
How to realize fully coupled simulation through MultiApps system in MOOSE

### DISCUSSION THREAD

**Discussion by sychang2021** [idaholab/moose#21545] [Q&A General] [answered]
I have a problem coupled by two systems which have twenty variables. When the number of mesh is large, solving the equations of the two systems simultaneously will face the limitation of solution speed and memory. So I think it is possible to solve the two problems separately (in two input files) and pass the information through the MultiApps system. But in doing so, is the solution method `tightly coupled` instead of `fully coupled` ? Can I achieve full coupling through the `MultiApps` system?

**Comment by GiudGiud** [comment, marked as answer]
We refer to full coupling as the single-matrix approach, where all variables are solved for in the same system.

You can only achieve tight or loose coupling using MultiApps. At convergence, the results from full and tight coupling should be the same

**Reply by sychang2021** [reply]
On my computer, I'm using Newton's method to solve a problem with 20 variables, 20,000 grids, which is a coupling of two different problems. Below is my setup in Executioner. 
```
  solve_type = 'NEWTON'
  petsc_options = '-snes_converged_reason -ksp_converged_reason -snes_linesearch_monitor'

  #petsc_options_iname = '-pc_type -sub_pc_type -ksp_gmres_restart -pc_gasm_overlap -sub_pc_factor_shift_type -pc_gasm_blocks -sub_pc_factor_mat_solver_type'
  #petsc_options_value = 'gasm     lu           200                1                NONZERO                   16             superlu_dist'

  petsc_options_iname = '-pc_type -pc_factor_shift_type'
  petsc_options_value = 'lu       NONZERO'
```
At the beginning, it takes 1000s to solve each step. When the solution reaches more than 30 steps, it takes more than 1 hour to solve each step. But when I solve the two subproblems separately, the total time to solve one step is about 200 seconds. So I thought it would be more feasible to solve them separately. But I'm also worried about the accuracy of the solution. Do you think fully coupled and tightly coupled results are the same when converging?

I would very much like to use the single-matrix approach to solving the problem, which looks wonderful and ideal. But limited by solving time and memory, maybe I have to take another approach.
I have tried other petsc setups like the other petsc setup above. The solution time of each step is about 1/3 of the direct lu method. But still too slow for a large problem. Do you have any suggestions?

**Reply by GiudGiud** [reply]
It takes 200s with fixed point iterations? Or did you just run one solve then the other (loose coupling).

Yes at convergence fully and tightly coupled match. I have yet to see a case where they dont.

The single matrix approach is simple but is not the most common method. Numerically, it's more often than not beneficial to separate solves to reduce the size of the system.

**Reply by sychang2021** [reply]
I have split a coupling problem into two separate problems. One of the problems took 130 seconds to solve in one step, and the other took 70 seconds to solve in one step. It looks like I need to use the MultiApps system for tight coupling. Thanks for your suggestion and reply.

**Reply by GiudGiud** [reply]
you need to iterate the coupling though. Like both problems need to run several times for each time step, exchanging information using Transfers.
See the fixed point parameters in the dropdown here:
https://mooseframework.inl.gov/source/multiapps/FullSolveMultiApp.html
